### PR TITLE
Fixes a flakiness in GPUDevicePlugin e2e test.

### PR DESCRIPTION
**Pull request by jiayingz**
Waits till nvidia gpu disappears from all nodes after deleting the
device plug DaemonSet to make sure its pods are deleted from all nodes.

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
https://github.com/kubernetes/kubernetes/issues/53281

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: test/e2e/scheduling/nvidia-gpus.go [hunk 1]
```diff
@@ -139,6 +139,24 @@ func areGPUsAvailableOnAllSchedulableNodes(f *framework.Framework) bool {
...
+                if node.Spec.Unschedulable {
+                        continue
+                }
+                framework.Logf("gpuResourceName %s", gpuResourceName)
+                if val, ok := node.Status.Capacity[gpuResourceName]; ok && val.Value() > 0 {
+                        framework.Logf("Nvidia GPUs available on Node: %q", node.Name)
+                        return true
+                }
+        }
+        framework.Logf("Nvidia GPUs exist on all schedulable nodes")
```
Comment (k82cn) [inline test/e2e/scheduling/nvidia-gpus.go]: should be "... does not exist ..."?

**Comment by k82cn** [comment]
The logic LGTM, but I'd like someone who knows GPU well to mark the labels :).

File: test/e2e/scheduling/nvidia-gpus.go [hunk 2]
```diff
@@ -222,8 +240,8 @@ var _ = SIGDescribe("[Feature:GPUDevicePlugin]", func() {
 		framework.Logf("Successfully deleted device plugin daemonset. Wait for resource to be removed.")
 		// Wait for Nvidia GPUs to be not available on nodes
```
Comment (mindprince) [inline test/e2e/scheduling/nvidia-gpus.go]: Change this to:
```
Wait for Nvidia GPUs to be unavailable on all nodes.
```

File: test/e2e/scheduling/nvidia-gpus.go (hunk 1 above)
Comment (mindprince) [reply test/e2e/scheduling/nvidia-gpus.go]: Yeah, change this to:
```
Nvidia GPUs don't exist on any schedulable node.
```

**Review by rohitagarwal003 — COMMENTED** [review]
Minor text updates. LGTM otherwise.

**Comment by mindprince** [comment]
Also need to run `gofmt`

File: test/e2e/scheduling/nvidia-gpus.go (hunk 1 above)
Comment (jiayingz) [reply test/e2e/scheduling/nvidia-gpus.go]: Good catch! Done.

File: test/e2e/scheduling/nvidia-gpus.go (hunk 2 above)
Comment (jiayingz) [reply test/e2e/scheduling/nvidia-gpus.go]: Done.

**Review by jiayingz — COMMENTED** [review]
Thanks a lot for the reviews! PTAL.

**Review by rohitagarwal003 — APPROVED** [review]
/lgtm

**Comment by vishh** [comment]
/approve
/lgtm

Thanks for the fix @jiayingz

**Comment by vishh** [comment]
@jiayingz can you create a bug to get this fix pulled back into v1.8 release branch?

**Comment by dims** [comment]
@jiayingz @vishh - submitted a cherry-pick. 

@vishh Can you please apply the appropriate labels? in #53290

**Pull request merged**